_posts/2019-05-23-quarkus-and-angular-ui-development-mode.adoc

```diff
@@ -0,0 +1,391 @@
+
+In case you are wondering what this is about, it is actually a really trimmed down version of something I needed to do do OAuth in my own application. Briefly, for my OAuth case, the flow is something along the lines of `/servlet/make-external-call` initiating the sign-in with the OAuth provider. This results in a few calls back and forth between the back-end and the OAuth provider, culiminating with the OAuth provider calling `/servlet/callback` with the authentication token. My servlet then caches the token, redirects to a route in the client which then makes a REST call to download the cached token.
+
+Next we populate our Angular application with our routes in `app-routing.module.ts`:
+[source,java]
+----
+const routes: Routes = [
+  {path: '', pathMatch: 'full', component: DefaultComponent},
+  {path: 'other', component: OtherComponent},
+  {path: 'rest', component: RestComponent},
+  {path: 'clientCallback', component: ClientCallbackComponent}
+];
+----
+
+We change `app.component.ts` to set up a simple application with the components listed above. The contents of the file can be found https://github.com/kabir/blog-quarkus-ui-development/blob/step-3/webapp/src/app/app.component.ts[here] (since it is a bit lengthy and not very interesting). Finally we update `app.module.ts` to declare our added components, and import the `HttpClientModule` which is needed by `RestComponent` to do its REST calls.
+
+[source,java]
+----
+@NgModule({
+  declarations: [
+    AppComponent,
+    DefaultComponent,
+    OtherComponent,
+    RestComponent,
+    ClientCallbackComponent
+  ],
+  imports: [
+    BrowserModule,
+    AppRoutingModule,
+    HttpClientModule
+  ],
+  providers: [],
+  bootstrap: [AppComponent]
+})
+export class AppModule { }
+
+----
+
+
+All this does is that when we go to the root of the application, we will go to DefaultComponent. `DefaultComponent` has routerLinks to `OtherComponent` and `RestComponent` (for routerLinks Angular does not hit the back-end), and a normal link to `/servlet/make-external-call`.
+
+If we go to `/other`, we end up in `OtherComponent` which just has a link back to `DefaultComponent`.
+
+If we go to `/rest`, we end up in `RestComponent` which displays data from the `SampleResource` we saw above and also has a link back to `DefaultComponent`.
+
+Finally if we go to `/clientCallback` (which is triggered via `/servlet/callback`  in `/SampleServlet` we end up in `ClientCallbackComponent`.
+
+====== Illustrating the problem
+Now package and start the application by running:
+[source,bash]
+----
+$mvn package quarkus:dev -Dui.dev
+----
+If you go to http://localhost:8080 you will get a page with links to `Other`, `Rest` and `Default`. Click on the `Other` and `Rest` ones, and it should all work.
+
+However while in the `Other` component, so that the address in the browser is http://localhost:8080/other, if you try to refresh the page you will end up with the following error message:
```

Comment:
This message look a bit strange. What about following version?

However while in the `Other` component (http://localhost:8080/other), if you try to refresh the page you will end up with the following error message: